### DISCUSSION TITLE
Dynamically respecting .gitignore

### DISCUSSION THREAD

**Discussion by miguel9554** [junegunn/fzf#3363] [Q&A]
As explained in [README](https://github.com/junegunn/fzf#respecting-gitignore), gitignore can be respected by setting as a default command `rg` for example.

With `rg`, `--no-ignore` allows to ignore `.gitignore`. How can I pass this flag to `rg` from `fzf`, if I've set `FZF_DEFAULT_COMMAND='rg --files'`?

**Comment by LangLangBart** [comment]
- If you have `FZF_DEFAULT_COMMAND='rg --files'` set in your `.bashrc`/`.zshrc`, you should be able to simply add a keybinding when running `fzf`, for example:

```sh
fzf --bind 'ctrl-a:reload:eval $FZF_DEFAULT_COMMAND --no-ignore'
# or
fzf --bind 'ctrl-a:reload:rg --files --no-ignore'
```

```man
man rg
       -u, --unrestricted ...
           Reduce the level of "smart" searching. A single -u won't respect
           .gitignore (etc.) files (--no-ignore). Two -u flags will
           additionally search hidden files and directories (-./--hidden).
           Three -u flags will additionally search binary files (--binary).
 
           rg -uuu is roughly equivalent to grep -r.
```

--- 

- If you want to switch between the two variants read the [ADVANCED.md](../blob/master/ADVANCED.md#toggling-between-data-sources).

**Reply by miguel9554** [reply]
@LangLangBart thanks for the answer, github does not notify so I've just seen it.

This wouldn't work for my use case, I actually want to control ignore of files from vim, using fzf.vim. I think the keybind approach would not work. I'm looking to pass some extra args to fzf, that would eventually be passed to rg. Maybe it's better to explore the solution in fzf.vim directly? Thanks
